src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -641,24 +687,26 @@ Result SpatialJoinAlgorithms::BoundingBoxAlgorithm() {
   for (size_t i = 0; i < otherResult->numRows(); i++) {
     auto geopoint = getPoint(otherResult, i, otherResJoinCol);
     std::vector<Box> bbox;
+    rtreeEntry entry{i, std::nullopt, std::nullopt, std::nullopt};
+    entry.row = i;
 
     if (!geopoint) {
-      try {
-        std::string areastring = betweenQuotes(
-            ExportQueryExecutionTrees::idToStringAndType(
-                qec_->getIndex(), otherResult->at(i, otherResJoinCol), {})
-                .value()
-                .first);
-        auto areaBox = calculateBoundingBoxOfArea(areastring);
-        auto midpoint = calculateMidpointOfBox(areaBox);
-        bbox = computeBoundingBox(
-            midpoint,
-            getMaxDistFromMidpointToAnyPointInsideTheBox(areaBox, midpoint));
-      } catch (...) {
-        printWarning();
+      AnyGeometry geometry;
+      if (!getAnyGeometry(otherResult, i, otherResJoinCol, geometry)) {
+        // nothing to do. When parsing a point or an area fails, a warning
+        // message gets printed at another place and the point/area just gets
+        // skipped
         continue;
       }
+      auto areaBox = boost::apply_visitor(BoundingBoxVisitor(), geometry);
+      entry.geometry = std::move(geometry);
+      auto midpoint = calculateMidpointOfBox(areaBox);
+      bbox = computeBoundingBox(
```

Comment:
This also looks very duplicaty, 
Please follow the following two rules:

1. No copying of code (hard to maintain)
2. No copying of large objects if it can be avoided (because of time and space wasteage)